File: DPBO-praktikum/4/python/SivitasAkademik.py
from Human import Human

class SivitasAkademik(Human):
    def __init__(self, nik, nama, jenis_kelamin, asal_universitas, email_edu):
        super().__init__(nik, nama, jenis_kelamin)
        self._asal_universitas = asal_universitas
        self._email_edu = email_edu

    def set_asal_universitas(self, asal_universitas):
        self._asal_universitas = asal_universitas

    def get_asal_universitas(self):
        return self._asal_universitas

    def set_email_edu(self, email_edu):
        self._email_edu = email_edu

    def get_email_edu(self):
        return self._email_edu

File: DPBO-praktikum/4/cpp/SivitasAkademik.cpp
// memanggil file header
#include "header.hh"

// constructor kosong
SivitasAkademik::SivitasAkademik(){

}

// constructor berparameter
SivitasAkademik::SivitasAkademik(string nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu)
: Human(nik, nama, jenis_kelamin) {
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
}

// mengisi nilai asal universitas
void SivitasAkademik::set_asal_universitas(string asal_universitas) {
    this->asal_universitas = asal_universitas;
}

// mengambil nilai universitas
string SivitasAkademik::get_asal_universitas() {
    return asal_universitas;
}

// mengisi nilai email
void SivitasAkademik::set_email_edu(string email_edu) {
    this->email_edu = email_edu;
}

// mengambil nilai email
string SivitasAkademik::get_email_edu() {
    return email_edu;
}

// destructor
SivitasAkademik::~SivitasAkademik(){

}
